sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -199,21 +195,26 @@ def task():
         "Currently using %s threads." % len(self._process_thread_pool._threads))
 
   def _request_process_bundle_split(self, request):
-    self._request_process_bundle_action(request)
+    self._request_process_bundle_action(request, self._progress_thread_pool)
+
+  def _request_finalize_bundle(self, request):
+    self._request_process_bundle_action(request, self._finalize_thread_pool)
```

Comment:
We can re-use process_thread_pool for this rather than creating another one.

Reply:
Just curious, why progress has its own thread_pool?

Reply:
The intent here is that progress requests and splits should never get queued behind finishing a work item.

Reply:
Thanks for explanation. Though I still don't get why the progress_pool can serve for the intent you mentioned. But I think the previous instrucion_id_vs_worker is to make sure the progress/split can only call on an active bundle.